backend/mcp_backend/routers/mcp.py

```
@@ -0,0 +1,963 @@
+async def create_mcp_client(url: str, headers: Optional[Dict] = None):
+    """Create FastMCP client connection to MCP server"""
+    try:
+        # FastMCP Client can connect to URLs, FastMCP servers, or stdio processes
+        client = Client(url)
+        return client
+    except Exception as e:
+        log.exception(f"Failed to create MCP client for {url}: {e}")
+        raise HTTPException(
+            status_code=400,
+            detail=f"MCP: Unable to connect to server at {url}. Please check the URL and try again.",
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)